### DISCUSSION TITLE
Is there a way to restart hls from the original m3u8 instead of the sub m3u8?

### DISCUSSION THREAD

**Discussion by favainhat** [streamlink/streamlink#5101] [Q&A] [answered]
In the video source I want to save, m3u8 expires when the stream is disconnected, so it needed to get the m3n8 address from the original m3u8 again.

in fetch_streams_with_retry

     while not streams:
         sleep (interval)

         try:
             streams = fetch_streams(plugin)
         except FatalPluginError:
             raise
         except PluginError as err:
             log.error(err)

         if count > 0:
             attempts += 1
             if attempts >= count:
                 break
A workaround I thought of was to reload the plugin in this fucntion of streamlink_cli, But I would like to know if such a feature exists or if there is other good way.

Or, in this case, should I immediately disconnect the stream, start streamlink again, and save the stream?

**Comment by bastimeyer** [comment, marked as answer]
Just restart streamlink, so it can fully resolve the stream URL again. Stream expirations usually need to be implemented in a plugin with a custom HLSStream subclass, so a seamless transition can be achieved.

The workaround you've mentioned doesn't make any sense, since the plugin was already correctly resolved from the input URL. It'd just need to re-fetch the stream from the same plugin instance. Re-fetching a stream via streamlink_cli once it was already resolved however won't work, because that would most certainly always result in a stream discontinuity of two separate streams, even though you're just "reloading"/"refreshing" the same stream. This is not supported.

**Reply by favainhat** [reply]
Yes. restart streamlink seems like right choice.
